File: benchmark/python/cumulative_sum.py
import time

n = 100000
arr = []
for i in range(1, n + 1):
    arr.append(i)
start = time.time() * 1000
cum = []
s = 0
for i in range(n):
    s = s + arr[i]
    cum.append(s)
end = time.time() * 1000
print(f"Last: {cum[n - 1]}")
print(f"Time: {end - start:.0f}ms")

File: benchmark/lua/cumulative_sum.lua
local n = 100000
local arr = {}
for i = 1, n do
    arr[i] = i
end
local start = os.clock()
local cum = {}
local s = 0
for i = 1, n do
    s = s + arr[i]
    cum[i] = s
end
local finish = os.clock()
print("Last: " .. cum[n])
print(string.format("Time: %.0fms", (finish - start) * 1000))

